KSP/src/main/kotlin/io/github/jan/supabase/ksp/SelectableSymbolProcessor.kt

```diff
@@ -0,0 +1,232 @@
+package io.github.jan.supabase.ksp
+
+import com.google.devtools.ksp.processing.CodeGenerator
+import com.google.devtools.ksp.processing.Dependencies
+import com.google.devtools.ksp.processing.KSPLogger
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.processing.SymbolProcessor
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.google.devtools.ksp.symbol.KSFile
+import com.google.devtools.ksp.symbol.KSNode
+import com.google.devtools.ksp.symbol.KSValueParameter
+import com.google.devtools.ksp.symbol.Modifier
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import io.github.jan.supabase.annotations.SupabaseInternal
+import io.github.jan.supabase.postgrest.Postgrest
+import io.github.jan.supabase.postgrest.annotations.ApplyFunction
+import io.github.jan.supabase.postgrest.annotations.Cast
+import io.github.jan.supabase.postgrest.annotations.ColumnName
+import io.github.jan.supabase.postgrest.annotations.Foreign
+import io.github.jan.supabase.postgrest.annotations.JsonPath
+import io.github.jan.supabase.postgrest.annotations.Selectable
+
+class SelectableSymbolProcessor(
+    private val codeGenerator: CodeGenerator,
+    private val logger: KSPLogger,
+    private val options: Map<String, String>
+) : SymbolProcessor {
+
+    val packageName = options["selectablePackageName"] ?: "io.github.jan.supabase.postgrest"
+    val fileName = options["selectableFileName"] ?: "PostgrestColumns"
+
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        val symbols = resolver.getSymbolsWithAnnotation(Selectable::class.java.name).filterIsInstance<KSClassDeclaration>()
+        if (!symbols.iterator().hasNext()) return emptyList()
+        val types = hashMapOf<String, String>()
+        symbols.forEach { symbol ->
+            val className = symbol.simpleName.asString()
+            val qualifiedName = symbol.simpleName.asString() //Kotlin JS doesn't support qualified names
```

Comment:
Name clashes could be possible due to only using the simple name, but we cannot use qualified names because of Kotlin JS support.